src/defrag.c

```diff
@@ -1088,8 +1109,11 @@ void activeDefragCycle(void) {
             db = &server.db[current_db];
             kvstoreDictLUTDefrag(db->keys, dictDefragTables);
             kvstoreDictLUTDefrag(db->expires, dictDefragTables);
+            defrag_state = DEFRAG_KEYS;
             cursor = 0;
             expires_cursor = 0;
+            pubsub_cursor = 0;
+            pubsubshard_cursor = 0;
```

Comment:
i wonder if we really need 4 cursors, or can we suffice with one?
preparing ahead of time for the next top level kvstore we'll want to handle here.

Reply:
cursor has been merged, but the optimization isn't yet complete, please wait for me to ping if it's ready.